# given the result of a tabletop detection, adds the objects to
# the collision map and returns them as instances of GraspableObjects
# with correct collision names

# a list of clusters and database models, directly as produced by the detector
# note that it is possible to not set the Table inside here (leave its
# extents set to zero), in which case it does not get added to the environment
tabletop_object_detector/TabletopDetectionResult detection_result

# whether the current collision models should be reset before adding new models
bool reset_collision_models

# whether the current list of models attached to the robot should be reset
bool reset_attached_models

# what tf frame the results should be returned in
# if empty, results will be in the same frame as the input 
string desired_frame

---

# the objects added to the collision environment and packaged as graspable objects
object_manipulation_msgs/GraspableObject[] graspable_objects

# the list of collision names that the objects got in the environment
# in the same order as graspable_objects
string[] collision_object_names

# the name that the table got in the collision map, if any
string collision_support_surface_name
